### DISCUSSION TITLE
Compatibility of Dom.eventTarget with Dom.node_like<'a>?

### DISCUSSION THREAD

**Discussion by Zeta611** [TheSpyder/rescript-webapi#130] [Q&A] [answered]
I'd like to write something liek `button->Element.contains(~child=event->Event.target))`, but I get the following error:
```rescript
  This has type:
    Dom.eventTarget (defined as Dom.eventTarget_like<Dom._baseClass>)
  But this function argument is expecting:
    Dom.node_like<'a> (defined as Dom.eventTarget_like<Dom._node<'a>>)

  The incompatible parts:
    Dom._baseClass vs Dom._node<'a>
```

Is there a way to make this work? I just sprinkled some `Obj.magic` here now---is creating a conversion function with `identity` the only way, or am I missing something here?

Thanks!

**Comment by TheSpyder** [comment, marked as answer]
Even in JS, assuming the shape of `event.target` can be unsafe. So in that spirit there are built-in unsafe identity functions:
https://github.com/TheSpyder/rescript-webapi/blob/18d9d553201d575fa8e194d3370298b556b28747/src/Webapi/Dom/Webapi__Dom__EventTarget.res#L1595-L1597

Which, in your case, I think translates to `event->Event.target->EventTarget.unsafeAsElement`.

**Reply by Zeta611** [reply]
Thank you for the answer!
